### PR TITLE
feat: make GCP operation tracker timeout configuration [DET-3182]

### PR DESCRIPTION
**Pull request by shiyuann**
## Description

When creating or terminating an instance, the GCP provider sends a request to the GCP API and the response will be a GCP operation. By tracking the GCP operation, we are able to know if the instance is successfully created/terminated. There is a timeout before we consider a GCP operation no longer effective and we might retry after hitting the timeout. This PR intends to make that timeout period configurable. This PR changes the code and the doc relates to it.

[DET-3182](https://determinedai.atlassian.net/browse/DET-3182)

## Test Plan

Manually test it with the following provisioner config and see if the timeout is configurable:
```
provisioner:
  agent_docker_network: host
  agent_docker_image: determinedai/determined-dev:determined-agent-d90294d941964f104b50204232002fa551daa648
  max_idle_agent_period: 1m
  **operation_timeout_period: 10s**
  provider: gcp
  boot_disk_size: 200
  boot_disk_source_image: projects/determined-ai/global/images/pedl-environments-4e98289
  network_interface:
    network: projects/determined-ai/global/networks/default
    subnetwork: projects/determined-ai/regions/us-central1/subnetworks/default
    external_ip: false
  instance_type:
    machine_type: n1-standard-8
    gpu_type: nvidia-tesla-v100
    gpu_num: 2
    preemptible: false
  max_instances: 2
```

### CODE REVIEW

File: master/internal/provisioner/gcp_config_test.go [hunk 1]
```diff
@@ -53,7 +54,8 @@ func TestUnmarshalGCPClusterConfig(t *testing.T) {
 		"gpu_type": "nvidia-tesla-v100",
 		"gpu_num": 2
 	},
-	"max_instances": 100
+	"max_instances": 100,
+    "operation_timeout_period": "5m"
```
Comment (aaron276h) [inline master/internal/provisioner/gcp_config_test.go:58]: seems to not be properly indented.

File: docs/reference/cluster-config.txt [hunk 2]
```diff
@@ -421,6 +421,11 @@ The master supports the following configuration settings:
     - ``max_instances``: Max number of Determined agent instances. Defaults
       to 5.
 
+    - ``operation_timeout_period``: The timeout period to tracking a GCP
```
Comment (aaron276h) [inline docs/reference/cluster-config.txt:424]: `The timeout period for ...`

**Comment by hoanghphan** [comment]
I tried testing this on a GCP cluster with master & agent images built from this PR but it doesn't look like the timeout is having any effect on the master starting new agents.

I set the timeout to 30m and ran an experiment with sleep 120 in the agent startup script, and the master still requested a second agent, well before the 30 minute mark.

@shiyuann can you try testing on a GCP cluster to confirm?

**Comment by shiyuann** [comment]
@hoanghphan This timeout will only be relevant to the GCP operation and how we identify our operations change the statuses of GCP instances. It will not affect how we consider an agent is up so the agent startup script won't affect anything. I guess the right way to test it is to use a very small timeout and you will notice the master logs say can't track a GCP operation due to timeout.

**Pull request merged**